In [8]:
import csv
import pandas as pd
import statistics

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
def ler_csv_e_mostrar_tabelas(nome_arquivo):

    # Lê o arquivo CSV e carrega os dados em um DataFrame do pandas
     df = pd.read_excel(nome_arquivo)
    # Exibe a descrição estatística das colunas
     descricao = df.describe()
    # Exibe informaçõe adicionais
     info = df.info()

     print("Descrição Estatística:")
     print(descricao)

     print("\nInformações Adicionais:")
     print(info)

caminho_arquivo = '/content/drive/Shareddrives/modulo_8/Dados em CSV IBGE/PDA_PNME_2017_2018_2019.xlsx'

ler_csv_e_mostrar_tabelas(caminho_arquivo)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6904 entries, 0 to 6903
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   NOME_UF             6904 non-null   object
 1   NOME_MUNICIPIO      6904 non-null   object
 2   ESFERA_GOV_ESCOLA   6904 non-null   object
 3   LOCALIZACAO_ESCOLA  6904 non-null   object
 4   INEP_ESCOLA         6904 non-null   int64 
 5   NOME_ESCOLA         6904 non-null   object
 6   QUANT_ALUNADO       6904 non-null   int64 
 7   VALOR_RECEBIDO      6904 non-null   object
dtypes: int64(2), object(6)
memory usage: 431.6+ KB
Descrição Estatística:
        INEP_ESCOLA  QUANT_ALUNADO
count  6.904000e+03    6904.000000
mean   2.511464e+07     150.858633
std    6.576994e+06     119.129398
min    1.100071e+07      25.000000
25%    2.115043e+07      80.000000
50%    2.606106e+07     120.000000
75%    2.915887e+07     180.000000
max    5.301426e+07    1924.000000

Informações Adicionais:

In [6]:
pda_pnme = pd.read_excel('/content/drive/Shareddrives/modulo_8/Dados em CSV IBGE/PDA_PNME_2017_2018_2019.xlsx')

print(pda_pnme)

     NOME_UF NOME_MUNICIPIO ESFERA_GOV_ESCOLA LOCALIZACAO_ESCOLA  INEP_ESCOLA  \
0         RO    PORTO VELHO          ESTADUAL             Urbana     11000708   
1         RO    PORTO VELHO         MUNICIPAL              Rural     11000864   
2         RO    PORTO VELHO          ESTADUAL             Urbana     11001160   
3         RO    PORTO VELHO          ESTADUAL             Urbana     11001496   
4         RO    PORTO VELHO          ESTADUAL             Urbana     11002255   
...      ...            ...               ...                ...          ...   
6899      DF       BRASILIA          ESTADUAL              Rural     53003047   
6900      DF       BRASILIA          ESTADUAL             Urbana     53005996   
6901      DF       BRASILIA          ESTADUAL             Urbana     53006011   
6902      DF       BRASILIA          ESTADUAL             Urbana     53012119   
6903      DF       BRASILIA          ESTADUAL             Urbana     53014260   

                           

In [12]:
pda_pnme.columns

Index(['NOME_UF', 'NOME_MUNICIPIO', 'ESFERA_GOV_ESCOLA', 'LOCALIZACAO_ESCOLA',
       'INEP_ESCOLA', 'NOME_ESCOLA', 'QUANT_ALUNADO', 'VALOR_RECEBIDO'],
      dtype='object')

In [13]:
pda_pnme.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6904 entries, 0 to 6903
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   NOME_UF             6904 non-null   object
 1   NOME_MUNICIPIO      6904 non-null   object
 2   ESFERA_GOV_ESCOLA   6904 non-null   object
 3   LOCALIZACAO_ESCOLA  6904 non-null   object
 4   INEP_ESCOLA         6904 non-null   int64 
 5   NOME_ESCOLA         6904 non-null   object
 6   QUANT_ALUNADO       6904 non-null   int64 
 7   VALOR_RECEBIDO      6904 non-null   object
dtypes: int64(2), object(6)
memory usage: 431.6+ KB


In [16]:
pda_pnme.nunique()

NOME_UF                 27
NOME_MUNICIPIO        1461
ESFERA_GOV_ESCOLA        2
LOCALIZACAO_ESCOLA       2
INEP_ESCOLA           6904
NOME_ESCOLA           6727
QUANT_ALUNADO          532
VALOR_RECEBIDO        4311
dtype: int64

In [19]:
pda_pnme['QUANT_ALUNADO'].mean()


150.85863267670916

In [20]:
pda_pnme['QUANT_ALUNADO'].median()

120.0

In [21]:
pda_pnme['VALOR_RECEBIDO']

0        R$ 4.120,00
1       R$ 54.840,00
2        R$ 8.556,00
3       R$ 18.492,00
4       R$ 12.420,00
            ...     
6899     R$ 8.136,00
6900    R$ 37.920,00
6901     R$ 8.316,00
6902    R$ 11.268,00
6903    R$ 37.743,00
Name: VALOR_RECEBIDO, Length: 6904, dtype: object

In [26]:
pda_pnme.isnull().sum()

NOME_UF                  0
NOME_MUNICIPIO           0
ESFERA_GOV_ESCOLA        0
LOCALIZACAO_ESCOLA       0
INEP_ESCOLA              0
NOME_ESCOLA              0
QUANT_ALUNADO            0
VALOR_RECEBIDO        6904
dtype: int64

In [27]:
pda_pnme.isna().sum()

NOME_UF                  0
NOME_MUNICIPIO           0
ESFERA_GOV_ESCOLA        0
LOCALIZACAO_ESCOLA       0
INEP_ESCOLA              0
NOME_ESCOLA              0
QUANT_ALUNADO            0
VALOR_RECEBIDO        6904
dtype: int64

 ## Tabela PDA_PNME

 - Coluna 'VALOR_RECEBIDO' possui 6904 nulos